services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -492,6 +493,27 @@ public AccessToken transformAccessToken(KeycloakSession session, AccessToken tok
         }
         return token;
     }
+
+    public AccessToken transformUserInfoAccessToken(KeycloakSession session, AccessToken token, RealmModel realm, ClientModel client, UserModel user,
```

Comment:
this method is more or less a copy of the `transformAccessToken` method with the only addition of the `UserInfoTokenMapper` handling. I think it would make sense to refactor this.